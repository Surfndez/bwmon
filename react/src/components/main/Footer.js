import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    footer: {
        margin: theme.spacing(1),
    },
}));

const Footer = () => {
    const classes = useStyles();

    return (
        <Typography variant="subtitle2" className={classes.footer}>
            All usage information is in gigabytes
        </Typography>
    );
};

export default Footer;
