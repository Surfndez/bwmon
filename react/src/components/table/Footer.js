import React from 'react';
import { TableHead, TableRow, TableCell } from '@material-ui/core';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                const text =
                    index === 0
                        ? 'Totals'
                        : cellInfo.footer
                        ? cellInfo.convert
                            ? cellInfo.convert(values[cellInfo.id])
                            : values[cellInfo.id]
                        : '';
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

export default Footer;
