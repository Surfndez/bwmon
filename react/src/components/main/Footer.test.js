import React from 'react';
import { create } from 'react-test-renderer';
import Footer from './Footer';

describe('Footer', () => {
    test('renders correctly', () => {
        const tree = create(<Footer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
