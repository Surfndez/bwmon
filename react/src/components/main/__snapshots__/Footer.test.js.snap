// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<h6
  className="MuiTypography-root makeStyles-footer-1 MuiTypography-subtitle2"
>
  All usage information is in gigabytes
</h6>
`;
